src/benchq/compilation/initialize_julia.py

```diff
@@ -0,0 +1,48 @@
+import juliapkg
+
+# juliapkg.json is undesirable because you have to put it in the same directory as
+# the script that calls it. So we have to check whether the dependencies are already
+# installed and if not, install them. When adding a new dependency, you have to
+# manually add it to the dependency_dict below as well as the juliapkg.add() call.
+required_julia_version = "1.9.3"
+dependency_dict = {
+    "julia": "^" + required_julia_version,
+    "packages": {
+        "JSON": {"uuid": "682c06a0-de6a-54ab-a142-c8b1cf79cde6", "version": "0.21"},
+        "Jabalizer": {
+            "uuid": "5ba14d91-d028-496b-b148-c0fbc366f709",
+            "version": "0.4.4",
+        },
+        "TimerOutputs": {
+            "uuid": "a759f4b9-e2f1-59dc-863e-4aeb61b1ea8f",
+            "version": "0.5.23",
+        },
+        "StatsBase": {
+            "uuid": "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91",
+            "version": "0.34.0",
+        },
+    },
+}
+if juliapkg.deps.load_cur_deps() != dependency_dict:
+    print(
+        "No suitable Julia installation detected. "
+        "Automatically installing Julia "
```

Comment:
Does this message make sense in the case where the user does have the correct Julia version, but doesn't have the correct packages?

Reply:
Fixed https://github.com/zapatacomputing/benchq/pull/133/commits/9511cb82cbf1436b7bef498458e355043cb32c96